rustls/examples/internal/bench_impl.rs

```diff
@@ -27,15 +28,123 @@ use rustls::{
 };
 
 pub fn main() {
-    let mut args = std::env::args();
-    if args.len() > 1 {
-        args.next();
-        selected_tests(args);
-    } else {
-        all_tests();
+    let args = Args::parse();
+
+    let options = Options {
+        work_multiplier: args.multiplier,
+    };
+
+    match args.command.as_ref() {
+        Some(Command::Bulk {
+            cipher_suite,
+            plaintext_size,
+            max_fragment_size,
+        }) => {
+            for param in lookup_matching_benches(cipher_suite).iter() {
+                bench_bulk(param, &options, *plaintext_size, *max_fragment_size);
+            }
+        }
+        Some(
+            Command::Handshake { cipher_suite }
+            | Command::HandshakeResume { cipher_suite }
+            | Command::HandshakeTicket { cipher_suite },
+        ) => {
+            let resume = ResumptionParam::from_subcommand(args.command.as_ref().unwrap());
+
+            for param in lookup_matching_benches(&cipher_suite).iter() {
+                bench_handshake(param, &options, ClientAuth::No, resume);
+            }
+        }
+        Some(Command::Memory {
+            cipher_suite,
+            count,
+        }) => {
+            for param in lookup_matching_benches(&cipher_suite).iter() {
+                bench_memory(param, *count);
+            }
+        }
+        Some(Command::ListSuites) => {
+            for bench in ALL_BENCHMARKS {
+                println!(
+                    "{:?} (key={:?} version={:?})",
+                    bench.ciphersuite, bench.key_type, bench.version
+                );
+            }
+        }
+        None => {
```

Comment:
This feels like a somewhat counterintuitive use of `Option`, maybe `Command::All` instead? The derived clap stuff looks great!

Reply:
Did that, though the underlying value is still an Option, `fn command()` drops `None` to `Command::AllTests` (suggestion from https://github.com/clap-rs/clap/issues/975#issuecomment-1888988717 )